src/main/java/com/agata/jeeshop/services/CartService.java

```diff
@@ -82,8 +82,11 @@ public void removeProduct(Long cartId, Long id) {
         for (CartItem item : cart.getItems()) {
             if (item.getId() == id)
                 cartItemRepository.delete(item);
-
-              //  throw new RuntimeException("Cart item with id: " + cart.getItems() + " not found");
         }
+        throw new RuntimeException("Cart item with id: " + cart.getItems() + " not found");
```

Comment:
Ten wyjątek spowoduje, że nigdy nie będzie działało usuwanie. Jeżeli już chcesz go tutaj rzucać to rzuć dopiero wtedy gdy żaden element z listy nie został usunięty. Innymi słowy musisz albo zakończyć metodę po pomyślnym usunięciu przedmiotyu z koszyka albo rzucić wyjątek tylko wtedy gdy wykryjesz, że faktycznie go usunęłaś.